### ISSUE TITLE
SelectManyMenu: add 508 keyboard support

### ISSUE THREAD

**Issue by movalz** [primefaces/primefaces#1435]
I'm missing keyboard support for tag p:selectManyMenu like it is present for h:selectManyListbox (only Firefox does it exactly in the described way with CTRL/SHIFT keys):
- when focus is reached via TAB key one of the rows (the first row or first selected row) should be highlighted with a dotted frame
- pressing UP or DOWN key should move the highlighted row up/down. If neither CTRL nor SHIFT is pressed the new highlighted row should be marked.
- If CTRL is pressed together with UP/DOWN the old selection should stay and user can add/remove the currently highlighted row to the selection via SPACE key.
- If SHIFT is pressed together with UP/DOWN the old selection is changed to be the range from the first selected row to the reached row (like when clicking with SHIFT+mouse on a row)
- POS1/END should jump to the beginning/end of the items list with or without changing the selected item(s) depending on CTRL/SHIFT keys.
- PAGE UP/PAGE DOWN should scroll to previous/next page if not all items are visible
- If a p:outputLabel has a defined accesskey and is assigned to the p:selectManyMenu via its 'for' attribute the selectManyMenu should get focus (like when using TAB key to reach it) if the accesskey is pressed

I'm using PrimeFaces 6.0-RC3

**Comment by tandraschko** [comment]
a PR is really welcome here

**Comment by DmitriyAl** [comment]
I'm not sure about current status of this issue but it seems that at least for 6.2 version of primefaces the issue with arrows combining SHIFT still exists. If the component under focus and a user clicks arrows navigation around items works. But if a user holds SHIFT simultaniously with arrows pressing the behavior is incorrect. For down arrow combining with SHIFT only 3 items can be selected and if the user continues click on down arrow - nothing occurs. But if the user presses up arrow combining with SHIFT - the behavior little different. When two items are selected if user clicks up arrow one more time another one new item will be selected but the lowest one will be unselected. Looks like snake which moves upward. Code snippet could be taken from primefaces showcase for selectManyMenu component (https://www.primefaces.org/showcase/ui/input/manyMenu.xhtml) the basic one.

**Comment by ajsnyde** [comment]
I can't select/deselect options on the SelectManyMenu through keyboard as shown on https://www.primefaces.org/showcase/ui/input/manymenu/basic.xhtml. I've tried using Chrome, Firefox, and IE. I'm using 6.2 locally for a project and I need it 508 compliant.
